plugins/content_distribution/providers/facebook/lib/FacebookDistributionProvider.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * @package plugins.facebookDistribution
+ * @subpackage lib
+ */
+class FacebookDistributionProvider extends ConfigurableDistributionProvider
+{
+	/**
+	 * @var FacebookDistributionProvider
+	 */
+	protected static $instance;
+	
+	protected function __construct()
+	{
+	}
+	
+	/**
+	 * @return FacebookDistributionProvider
+	 */
+	public static function get()
+	{
+		if(!self::$instance)
+			self::$instance = new FacebookDistributionProvider();
+			
+		return self::$instance;
+	}
+	
+	/* (non-PHPdoc)
+	 * @see IDistributionProvider::getType()
+	 */
+	public function getType()
+	{
+		return FacebookDistributionPlugin::getDistributionProviderTypeCoreValue(FacebookDistributionProviderType::FACEBOOK);
+	}
+	
+	/* (non-PHPdoc)
+	 * @see IDistributionProvider::getName()
+	 */
+	public function getName()
+	{
+		return 'Facebook';
+	}
+
+	/* (non-PHPdoc)
+	 * @see IDistributionProvider::isDeleteEnabled()
+	 */
+	public function isDeleteEnabled()
+	{
+		return false;
+	}
+
+	/* (non-PHPdoc)
+	 * @see IDistributionProvider::isUpdateEnabled()
+	 */
+	public function isUpdateEnabled()
+	{
+		return true;
+	}
+
+	/* (non-PHPdoc)
+	 * @see IDistributionProvider::isMediaUpdateEnabled()
+	 */
+	public function isMediaUpdateEnabled()
+	{
+		return false;
+	}	
+	
+	/* (non-PHPdoc)
+	 * @see IDistributionProvider::isReportsEnabled()
+	 */
+	public function isReportsEnabled()
+	{
+		return false; // TODO - check if reports supported
+	}
+
+	/* (non-PHPdoc)
+	 * @see IDistributionProvider::isScheduleUpdateEnabled()
+	 */
+	public function isScheduleUpdateEnabled()
+	{
+		return false;
+	}
+
+	/* (non-PHPdoc)
+	 * @see IDistributionProvider::useDeleteInsteadOfUpdate()
+	 */
+	public function useDeleteInsteadOfUpdate()
+	{
+		return false;
+	}
+	
+	/* (non-PHPdoc)
+	 * @see IDistributionProvider::isAvailabilityUpdateEnabled()
+	 */
+	public function isAvailabilityUpdateEnabled()
+	{
+		return true;
```

Comment:
Availability update is not supported by the engine, in order to support availability update you must implement IDistributionEngineEnable and IDistributionEngineDisable.